lib/patch-series.ts

```diff
@@ -938,6 +946,7 @@ export class PatchSeries {
         const args = [
             "format-patch", "--thread", "--stdout", "--signature=ffmpeg-codebot",
             "--add-header=Fcc: Sent",
+            "--signoff", "--zero-commit",
```

Comment:
![](https://user-images.githubusercontent.com/4985349/191353418-e7bbf686-5484-4ba1-998d-ed5ef8e9967e.png)

The local git identity where this runs is ffmpegagent@gmail.com, so it would add that to all messages.

What would be the purpose of adding `--zero-commit`?

Reply:
It inserts the local identity, which we later replace with the original author identity before sending.

`--zero-commit` is because we'll be modifying the commit message, which means any git hash is invalidated, so it's more correct to set it to all-zeroes. This doesn't really matter for most purposes.

Reply:
GGG is using commit ids to relate things together like for example syncing comments on the ML back to the GitHub PR.

For example here: https://github.com/ffstaging/gitgitgadget/blob/fbf571c674fc39c7627a5dde9cd35e8524a9ae2b/lib/github-glue.ts#L190-L200